sql/src/main/java/io/crate/analyze/relations/AbstractTableRelation.java

```diff
@@ -82,7 +83,8 @@ protected Reference checkNestedArray(ColumnIdent ci, Reference reference) {
         Reference tmpRI = reference;
 
         while (!tmpCI.isColumn() && hasMatchingParent(tmpRI, IS_OBJECT_ARRAY)) {
-            if (DataTypes.isCollectionType(reference.valueType())) {
+            if (DataTypes.isCollectionType(reference.valueType()) &&
+                Schemas.isDefaultOrCustomSchema(reference.ident().tableIdent().schema())) {
```

Comment:
why is this okay for sys tables but not for user tables?

Reply:
afaik this limitation is due to lucene queries, right? happy to discuss that we leave the limitation also for sys tables. then we'd need to come up with a different schema for the allocations table, because otherwise `SELECT decisions['explanations'] FROM sys.allocations` would not be possible.

Reply:
I think the schema is fine. But would be great if you could try to figure why it doesn't work for doc tables and update the TODO